superset/db_engine_specs/hive.py

```diff
@@ -542,11 +542,9 @@ def update_impersonation_config(
             connect_args["configuration"] = configuration
 
     @staticmethod
-    def execute(  # type: ignore
-        cursor, query: str, async_: bool = False
-    ):  # pylint: disable=arguments-differ
-        kwargs = {"async": async_}
-        cursor.execute(query, **kwargs)
+    def execute(cursor, query: str):
```

Comment:
I have the latest python version and I'm getting errors with this kwarg not being recognized.

Reply:
```
raise SupersetErrorsException(superset-f79cd4889-88xps superset.exceptions.SupersetErrorsException: [SupersetError(message="databricks error: execute() got an unexpected keyword argument 'async'", error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'Databricks Interactive Cluster', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})] 
```

Reply:
What package and what version of that package should be required for this to work? This only happens when I try loading Unity Catalog data from Databricks in SQL Lab.

Reply:
My PyHive is up to date (`0.6.5`) and python version is `3.9.17`

Reply:
I've seen a similar issue for databricks, but not hive though.
@alldoami  - it would be useful to replicate the bug in the test, as I recall we have presto & hive running in the CI.

The issue with `async` - is that it is a reserved word in python:
```In Python 3.7 async became a keyword; you can use async_ instead:```

https://github.com/dropbox/PyHive/blob/d4ae481675ac5588ba9101596fa26f22ef0e77c4/README.rst#L55

Reply:
This is the URI that's failing me rn: `databricks://token:{access_token}@{server_hostname}:{port}/?&catalog={catalog}&schema={schema}`. It recognizes the catalogs in SQLLab but then fails to load any data because of the above error I posted.

Reply:
I also put more detail in the issue I posted: https://github.com/apache/superset/issues/24786

Reply:
@bkyryliuk any thoughts? Have you tried connecting to Databricks Unity Catalog before and succeeded?

Reply:
following up again... @bkyryliuk @john-bodley

Reply:
hm, `databricks://token` - that should not be using pyhive but rather native databricks sqlalchemy adapter. 
I haven't had a time to debug this issue and we may get to it in a couple weeks
cc @zhaorui2022 

consider flagging this to https://github.com/databricks/databricks-sql-python as well, I think it is coming from databricks connector

Reply:
<img width="1001" alt="image" src="https://github.com/apache/superset/assets/16328889/19c3a2a3-983f-4880-b71d-1fe7caed47f0">
current error

Reply:
This is also happening for others: https://apache-superset.slack.com/archives/C0170U650CQ/p1690793155124269

Reply:
Yeah, I've seen similar error when testing databricks integration. By the way https://github.com/apache/superset/blob/b370f91cd9e239224b3174436cee889421008cb2/superset/db_engine_specs/databricks.py#L149 should be used instead of the presto engine.

Reply:
```
consider flagging this to https://github.com/databricks/databricks-sql-python as well, I think it is coming from databricks connector
```

What is coming from the databricks connector?

Reply:
I assume so. By the way try to use `databricks+connector` in the connection string. Sqllab seems to be working well in our test deployment

Reply:
When I use `databricks+connector`, Unity Catalog data is not rendered in the drop down on the left, only the hive metastore data. I am able to query unity catalog data, but you have to know the exact name of the schema and table from unity catalog. Does that make sense?

Reply:
`databricks+connector` works for me for schema selection as well. this is in 3.0.0.rc3 version

Reply:
@bkyryliuk are you able to see Unity Catalog schemas? Not sure hive metastore schemas?

Reply:
Yes, I need to specify catalog in the connection string and see the schemas in the catalog:
conn string
```
databricks+connector://token:{token}@{host}:{port}/{catalog}?catalog={catalog}
```

```
{"connect_args":{"http_path":"/sql/1.0/warehouses/{id}","catalog":"{catalog}"}}
```

I got a bit lazy and did not fully figure out what setting is actually setting the catalog and I think that has changed over time with databricks connector

Reply:
Ah this works! I was getting both a little wrong - Is it possible to update the documentation on this so that others can connect to unity catalog?

Reply:
@sfirke